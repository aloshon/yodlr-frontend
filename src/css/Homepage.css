.header {
    font-size: 250%;
}

.btn {
    margin: 30px;
}