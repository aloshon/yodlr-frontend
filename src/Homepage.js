import React, {useContext} from "react";
import {useHistory} from "react-router-dom";
import Container from "react-bootstrap/Container";
import JumboTron from "react-bootstrap/Jumbotron";
import Button from "react-bootstrap/Button";
import "./css/Homepage.css";
import StateContext from "./StateContext";

/** 
* Homepage renders the Yodlr title and user info.
* If user is admin, load button for admin page to view all users
*/
const Homepage = ({user}) => {
    const history = useHistory();
    const {updateState} = useContext(StateContext);

    // signout user
    const signout = () => {
        localStorage.setItem("user", null);
        updateState({...user, state: "pending"});
        history.push("/register")
    }

    return (
        <Container>
            <JumboTron style={{marginBottom: "100px"}}>
                <h1 className="header">Yodlr Design</h1>
                <h2>{user.firstName} {user.lastName}</h2>
                <h3>{user.email}</h3>
                <h4 style={{color: user.state === "active" ? "green" : "red"}}>
                    {user.state}
                </h4>
                {user.isAdmin && <Button 
                size="lg" 
                variant="primary" 
                className="btn btn-primary"
                onClick={() => {history.push("/admin")}}
                >
                    Admin Page
                </Button>}
            </JumboTron>
            <Button 
            variant="secondary"
            style={{position: "absolute", left: 0, bottom: 0}}
            onClick={signout}>
                Sign Out
            </Button>
        </Container>
    )
};

export default Homepage