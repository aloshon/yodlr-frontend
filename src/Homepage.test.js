import { fireEvent, render } from '@testing-library/react';
import Homepage from './Homepage';
import {Router} from 'react-router-dom';
import { createMemoryHistory } from 'history';
import StateContext from "./StateContext";

describe("homepage should be rendered", () => {
    const history = createMemoryHistory();
    const user = {
        id: 1,
        firstName: "test",
        lastName: "testing",
        email: "test@gmail.com",
        state: "active",
        isAdmin: "true",
    }

    const {asFragment, getByText} = render(
        <Router history={history}>
            <StateContext.Provider value={{updateState: jest.fn()}}>
                <Homepage user={user} />
            </StateContext.Provider>
        </Router>
    );


    test('Load user data and admin page if applicable, clicking either button updates history', () => {
        
        const logo = getByText("Yodlr Design");
        const signout = getByText("Sign Out");
        expect(logo).toBeInTheDocument();
        // useHistory does not actually change the page in testing,
        // so we just check  path has changed and there are more
        // pages in history 
        expect(history.length).toBe(1);
        fireEvent.click(signout);
        expect(history.length).toBe(2);
    });
    
    test("matches snapshot", () => {
        expect(asFragment()).toMatchSnapshot();
    });
});