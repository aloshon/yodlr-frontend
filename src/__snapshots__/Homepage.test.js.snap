// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homepage should be rendered matches snapshot 1`] = `<DocumentFragment />`;
